llvm/lib/CodeGen/MachineTraceMetrics.cpp

```diff
@@ -761,6 +762,60 @@ static void updatePhysDepsDownwards(const MachineInstr *UseMI,
   }
 }
 
+/// Estimates the number of cycles elapsed between DefMI and UseMI, DefMI
+/// inclusive and UseMI exclusive, if they're non-null and in the same
+/// BasicBlock. Returns std::nullopt when UseMI is in a different MBB than
+/// DefMI.
+static std::optional<unsigned>
+estimateDefUseCycles(const TargetSchedModel &Sched, const MachineInstr *DefMI,
+                     const MachineInstr *UseMI) {
+  if (!DefMI || !UseMI || DefMI == UseMI)
+    return 0;
```

Comment:
Why would any of these cases happen? Particularly the DefMI == UseMI case

Reply:
I don't see how this is more "complete". Don't add checks for situations that should not occ

Reply:
Thanks, this makes sense to me after some reflection.